lib/src/Uncertainty/Distribution/GeneralizedExtremeValueFactory.cxx

```diff
@@ -82,6 +94,899 @@ GeneralizedExtremeValue GeneralizedExtremeValueFactory::buildAsGeneralizedExtrem
   return FittingTest::BestModelBIC(sample, factoryCollection, bic);
 }
 
+class GeneralizedExtremeValueLikelihoodEvaluation : public EvaluationImplementation
+{
+public:
+  GeneralizedExtremeValueLikelihoodEvaluation(const Sample & sample)
+  : EvaluationImplementation()
+  , sample_(sample)
+  {
+  }
+
+  GeneralizedExtremeValueLikelihoodEvaluation * clone() const override
+  {
+    return new GeneralizedExtremeValueLikelihoodEvaluation(*this);
+  }
+
+  UnsignedInteger getInputDimension() const override
+  {
+    return 3;
+  }
+
+  UnsignedInteger getOutputDimension() const override
+  {
+    return 1;
+  }
+
+  Point operator() (const Point & parameter) const override
+  {
+    const UnsignedInteger size = sample_.getSize();
+    const Scalar mu = parameter[0];
+    const Scalar sigma = parameter[1];
+    const Scalar xi = parameter[2];
+
+    // beware we cannot write -size, we need to cast to float first
+    Scalar ll = -1.0 * size * std::log(sigma);
+
+    for (UnsignedInteger i = 0; i < size; ++ i)
+    {
+      const Scalar yi = (sample_(i, 0) - mu) / sigma;
+      if (std::abs(xi) < SpecFunc::Precision)
+      {
+        ll += -yi - std::exp(yi);
+      }
+      else
+      {
+        const Scalar c1 = 1.0 + xi * yi;
+        if (c1 <= SpecFunc::Precision) // can be slightly off
+          return Point(1, -SpecFunc::MaxScalar);
+        ll += -(1.0 + 1.0 / xi) * std::log(c1) - std::pow(c1, -1.0 / xi);
+      }
+    }
+    return Point(1, ll);
+  }
+
+private:
+  Sample sample_;
+};
+
+
+static Scalar GeneralizedExtremeValueFactoryPWM(const Sample & sample, UnsignedInteger r)
+{
+  Scalar s = 0.0;
+  const UnsignedInteger size = sample.getSize();
+  for (UnsignedInteger i = r; i < size; ++ i)
+    s += std::exp(SpecFunc::LogGamma(1.0 * (i + 1)) - SpecFunc::LogGamma(1.0 * (i + 1 - r))) * sample(i, 0);
+  return s / std::exp(SpecFunc::LogGamma(size + 1.0) - SpecFunc::LogGamma(1.0 * (size - r)));
+}
+
+
+DistributionFactoryLikelihoodResult GeneralizedExtremeValueFactory::buildMethodOfLikelihoodMaximizationEstimator(const Sample & sample) const
+{
+  if (sample.getSize() < 3)
+    throw InvalidArgumentException(HERE) << "Error: cannot build a GeneralizedExtremeValue distribution from a sample of size < 3";
+  if (sample.getDimension() != 1)
+    throw InvalidArgumentException(HERE) << "Error: can build a GeneralizedExtremeValue distribution only from a sample of dimension 1, here dimension=" << sample.getDimension();
+
+  const Function objective(new GeneralizedExtremeValueLikelihoodEvaluation(sample));
+  OptimizationProblem problem(objective);
+  problem.setMinimization(false);
+
+  const Scalar zmin = sample.getMin()[0];
+  const Scalar zmax = sample.getMax()[0];
+  const Scalar mean = sample.computeMean()[0];
+
+  // sigma > 0
+  const Point lowerBound({-SpecFunc::MaxScalar, 0.0, -SpecFunc::MaxScalar});
```

Comment:
0.0 is not enough to enforce sigma>0. It should be either a small value taken from ResourceMap or SpecFunc::Epsilon